try_signal.hpp

```diff
@@ -142,31 +142,21 @@ namespace sig {
 
 #else
 	// windows
-
 	namespace detail {
-		void se_translator(unsigned int, _EXCEPTION_POINTERS* ExceptionInfo);
-
-		struct scoped_se_translator
-		{
-			scoped_se_translator()
-			{ _prev_fun = _set_se_translator(se_translator); }
-
-			~scoped_se_translator()
-			{ _set_se_translator(_prev_fun); }
-
-			scoped_se_translator(scoped_se_translator const&) = delete;
-			scoped_se_translator& operator=(scoped_se_translator const&) = delete;
-
-		private:
-			void (*_prev_fun)(unsigned int, struct _EXCEPTION_POINTERS*);
-		};
+		sig::errors::error_code_enum map_exception_code(DWORD);
 	}
 
 	template <typename F, typename... Args>
 	auto try_signal(F&& f, Args... args) -> decltype(f(args...))
 	{
-		detail::scoped_se_translator scope;
-		return f(args...);
+		__try
+		{
+			return f(args...);
+		}
+		__except (EXCEPTION_EXECUTE_HANDLER)
+		{
+			throw std::system_error(detail::map_exception_code(GetExceptionCode()));
```

Comment:
I was under the impression that C++ exceptions don't necessarily mix well with structured exceptions. I can't recall any references right now though (and I may have gotten this impression from attempts to support this in mingw).

One possible reason to prefer ``set_se_translator`` is that the earlier the exception is converted to a "real" exception, the earlier the stack frames are properly unwound (with destructors actually being called). I don't know that there's a difference, but it seems like it might. i.e. SEH don't cause destructors to be called when unwinding the stack.